File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/group/DiscoverProgressModifier.java
package de.cubbossa.pathfinder.group;

import de.cubbossa.pathfinder.misc.Named;
import de.cubbossa.pathfinder.misc.NamespacedKey;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

public interface DiscoverProgressModifier extends Modifier, Named {

  NamespacedKey KEY = NamespacedKey.fromString("pathfinder:discover-progress");

  @Override
  default NamespacedKey getKey() {
    return KEY;
  }

  NamespacedKey getOwningGroup();

  CompletableFuture<Double> calculateProgress(UUID playerId);
}

File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/group/DiscoverProgressModifier.kt
package de.cubbossa.pathfinder.group

import de.cubbossa.pathfinder.misc.Named
import de.cubbossa.pathfinder.misc.NamespacedKey
import de.cubbossa.pathfinder.misc.NamespacedKey.Companion.fromString
import java.util.*

interface DiscoverProgressModifier : Modifier, Named {

    val owningGroup: NamespacedKey

    suspend fun calculateProgress(playerId: UUID): Double

    companion object {
        val key: NamespacedKey = fromString("pathfinder:discover-progress")
    }
}
